### PR TITLE
Llapi updates

### PR DESCRIPTION
**Pull request by DvirDukhan**
This PR allows:
1. Retrieve model, script, and tensor Redis types from LLAPI so other modules can read/write AI keys 
2. Added LLAPI for variadic script input - modified the existing input and variadic input for integrity verification before adding new input. Added failure test for double variadic input.

### CODE REVIEW

File: src/script.h [hunk 1]
```diff
@@ -67,9 +67,25 @@ RAI_ScriptRunCtx* RAI_ScriptRunCtxCreate(RAI_Script* script,
  *
  * @param sctx input RAI_ScriptRunCtx to add the input tensor
  * @param inputTensor input tensor structure
+ * @param err error data structure to store error message in the case of
+ * failures
  * @return returns 1 on success ( always returns success )
```
Comment (filipecosta90) [inline src/script.h:72]: we should update the @return statement

File: src/script.h [hunk 2]
```diff
@@ -67,9 +67,25 @@ RAI_ScriptRunCtx* RAI_ScriptRunCtxCreate(RAI_Script* script,
...
+ * For each Allocates a RAI_ScriptCtxParam data structure, and enforces a shallow copy of
+ * the provided input tensor, adding it to the input tensors array of the
+ * RAI_ScriptRunCtx.
+ *
+ * @param sctx input RAI_ScriptRunCtx to add the input tensor
+ * @param inputTensors input tensors array
+ * @param len input tensors array len
+ * @param err error data structure to store error message in the case of
+ * failures
+ * @return returns 1 on success ( always returns success )
```
Comment (filipecosta90) [inline src/script.h:86]: same as above

File: src/script.c [hunk 3]
```diff
@@ -309,10 +333,7 @@ int RedisAI_Parse_ScriptRun_RedisCommand(RedisModuleCtx *ctx,
                         return -1;
                     }
                 }
-                if (!RAI_ScriptRunCtxAddInput(*sctx, inputTensor)) {
-                    RedisAI_ReplyOrSetError(ctx, error, RAI_ESCRIPTRUN, "ERR Input key not found");
-                    return -1;
-                }
+                if (!RAI_ScriptRunCtxAddInput(*sctx, inputTensor, error)) return -1;
```
Comment (lantiga) [inline src/script.c:336]: Why did we remove the input key error here?

**Comment by lantiga** [comment]
@filipecosta90 I'm ok to merge, but in perspective (sooner than later) we should have a C++ test suite

**Pull request merged**